.github/workflows/update.yml

```diff
@@ -25,35 +25,76 @@ jobs:
             branch: 'nixos-24.05'
           - selectedBranch: 'nixos-24.05'
             branch: main
+
     name: Update the flake inputs and generate options
     runs-on: ubuntu-latest
     timeout-minutes: 40
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          token: ${{ secrets.GH_TOKEN_FOR_UPDATES }}
-          ref: '${{ matrix.branch }}'
+          ref: ${{ matrix.branch }}
 
       - name: Install Nix
         uses: cachix/install-nix-action@v26
         with:
           nix_path: nixpkgs=channel:nixos-unstable
           github_access_token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Configure git
+        run: |
+          git config user.name 'github-actions[bot]'
+          git config user.email '41898282+github-actions[bot]@users.noreply.github.com'
+
       - name: Update flake.lock
-        id: update
-        uses: DeterminateSystems/update-flake-lock@v21
+        run: nix flake update --commit-lock-file
+
+      - name: Update autogenerated files
+        run: nix run .#generate-files
+
+      - name: Create Pull Request
+        id: pr
+        uses: peter-evans/create-pull-request@v6
         with:
-          pr-title: "Update flake.lock (${{ matrix.branch }})"
+          # TODO: use an SSH deploy key instead
+          # See https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs
+          # and https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#push-using-ssh-deploy-keys
```

Comment:
This one we could leave for now.

Reply:
I'm not going to test the PR works until over the weekend, so I don't mind including it in this PR. Doesn't matter too much either way though, don't mind waiting until it's merged.